workbench-core/example/infrastructure/integration-tests/tests/datasets/accessPermissions.test.ts

```diff
@@ -315,4 +314,171 @@ describe('DataSets access permissions integration tests', () => {
+            identityType: 'GROUP',
+            identity: groupId,
+            accessLevel: 'read-write'
+          }
+        })
+      ).resolves.toMatchObject({
+        data: {
+          dataSetId: dataSetId,
+          permissions: [
+            {
+              identityType: 'GROUP',
+              identity: groupId,
+              accessLevel: 'read-write'
+            }
+          ]
+        }
+      });
+    });
+    it('removes access permissions for a user.', async () => {
+      const createDataSetResponse = await adminSession.resources.datasets.create({}, true);
+      const dataSetId: string = createDataSetResponse.data.id;
+      await (adminSession.resources.datasets.children.get(dataSetId) as Dataset).addAccess({
+        permission: {
+          identityType: 'USER',
+          identity: userId,
+          accessLevel: 'read-only'
+        }
+      });
+      await expect(
+        (adminSession.resources.datasets.children.get(dataSetId) as Dataset).removeAccess({
+          permission: {
+            identityType: 'USER',
+            identity: userId,
+            accessLevel: 'read-only'
+          }
+        })
+      ).resolves.toMatchObject({
+        data: {
+          dataSetId: dataSetId,
+          permissions: [
+            {
+              identityType: 'USER',
+              identity: userId,
+              accessLevel: 'read-only'
+            }
+          ]
+        }
+      });
+    });
+    it('throws if "identityType" is not "GROUP" or "USER"', async () => {
+      const createDataSetResponse = await adminSession.resources.datasets.create({}, true);
+      const dataSetId: string = createDataSetResponse.data.id;
+      const createGroupResponse = await adminSession.resources.groups.create({}, true);
+      const { groupId } = createGroupResponse.data;
+      await (adminSession.resources.datasets.children.get(dataSetId) as Dataset).addAccess({
+        permission: {
+          identityType: 'GROUP',
+          identity: groupId,
+          accessLevel: 'read-only'
+        }
+      });
+      await expect(
+        (adminSession.resources.datasets.children.get(dataSetId) as Dataset).removeAccess({
+          permission: {
+            identityType: mockBadValue,
+            identity: groupId,
+            accessLevel: 'read-only'
+          }
+        })
+      ).rejects.toThrow(new HttpError(400, {}));
+    });
+    it('throws if "accessLevel" is not "read-only" or "read-write"', async () => {
+      const createDataSetResponse = await adminSession.resources.datasets.create({}, true);
+      const dataSetId: string = createDataSetResponse.data.id;
+      const createGroupResponse = await adminSession.resources.groups.create({}, true);
+      const { groupId } = createGroupResponse.data;
+      await (adminSession.resources.datasets.children.get(dataSetId) as Dataset).addAccess({
+        permission: {
+          identityType: 'GROUP',
+          identity: groupId,
+          //@ts-ignore
```

Comment:
do we need this `@ts-ignore`?

Reply:
yes, testing what happens when a non-ts user passes bad input.

Reply:
it looks like access level is a correct type though?

Reply:
If you are trusting the TS Compiler to validate your inputs, you are leaving yourself vulnerable.

1. Package consumers are not bound to TypeScript. TypeScript just compile to JavaScript.
2. Type checking is compile-time, not run-time. This is a run-time test.

`@ts-ignore` is necessary in this TEST file to enable a run-time level test.

Reply:
I understand that. In this case it isnt doing anything since `accessLevel` is `'read-only'` which is explicitly allowed, so it is unnecessary to disable type checking